diff --git a/app/Resources/views/easy_admin/list.html.twig b/app/Resources/views/easy_admin/list.html.twig
index 36f01c6..c3ec533 100644
--- a/app/Resources/views/easy_admin/list.html.twig
+++ b/app/Resources/views/easy_admin/list.html.twig
@@ -14,3 +14,20 @@
 
     {{ parent() }}
 {% endblock item_actions %}
+
+{% block global_actions %}
+    {{ parent() }}
+
+    {% if easyadmin_action_is_enabled_for_list_view('export', _entity_config.name) %}
+        {% set _action = easyadmin_get_action_for_list_view('export', _entity_config.name) %}
+
+        {% block export_action %}
+            <div class="button-action">
+                <a class="{{ _action.css_class|default('') }}" href="{{ path('easyadmin', _request_parameters|merge({ action: _action.name })) }}" target="{{ _action.target }}">
+                    {% if _action.icon %}<i class="fa fa-{{ _action.icon }}"></i>{% endif %}
+                    {{ _action.label is defined and not _action.label is empty ? _action.label|trans(_trans_parameters) }}
+                </a>
+            </div>
+        {% endblock export_action %}
+    {% endif %}
+{% endblock global_actions %}
diff --git a/app/config/config.yml b/app/config/config.yml
index 5709bfd..dd337d8 100644
--- a/app/config/config.yml
+++ b/app/config/config.yml
@@ -92,7 +92,9 @@ easy_admin:
             field_id: 'admin/fields/_id.html.twig'
     list:
         title: 'List of %%entity_label%%'
-        actions: ['show']
+        actions:
+            - 'show'
+            - { name: 'export', icon: 'download', css_class: 'btn btn-primary' }
     show:
         max_results: 6
     entities:
diff --git a/src/AppBundle/Controller/EasyAdmin/AdminController.php b/src/AppBundle/Controller/EasyAdmin/AdminController.php
index 1c6499a..adcce42 100644
--- a/src/AppBundle/Controller/EasyAdmin/AdminController.php
+++ b/src/AppBundle/Controller/EasyAdmin/AdminController.php
@@ -2,6 +2,7 @@
 
 namespace AppBundle\Controller\EasyAdmin;
 
+use AppBundle\Entity\Genus;
 use JavierEguiluz\Bundle\EasyAdminBundle\Controller\AdminController as BaseAdminController;
 
 class AdminController extends BaseAdminController
@@ -21,4 +22,30 @@ class AdminController extends BaseAdminController
             'id' => $id,
         ]);
     }
+
+    public function exportAction()
+    {
+        throw new \RuntimeException('Action for exporting an entity not defined');
+    }
+
+    public function exportGenusAction()
+    {
+        $sortDirection = $this->request->query->get('sortDirection');
+        if (empty($sortDirection) || !in_array(strtoupper($sortDirection), ['ASC', 'DESC'])) {
+            $sortDirection = 'DESC';
+        }
+
+        $queryBuilder = $this->createListQueryBuilder(
+            $this->entity['class'],
+            $sortDirection,
+            $this->request->query->get('sortField'),
+            $this->entity['list']['dql_filter']
+        );
+
+        return $this->get('app.csv_exporter')->getResponseFromQueryBuilder(
+            $queryBuilder,
+            Genus::class,
+            'genuses-export.csv'
+        );
+    }
 }
diff --git a/src/AppBundle/Twig/Extension/EasyAdminExtension.php b/src/AppBundle/Twig/Extension/EasyAdminExtension.php
index 79dd041..0e004eb 100644
--- a/src/AppBundle/Twig/Extension/EasyAdminExtension.php
+++ b/src/AppBundle/Twig/Extension/EasyAdminExtension.php
@@ -37,6 +37,10 @@ class EasyAdminExtension extends EasyAdminTwigExtension
 
         $actions = parent::getActionsForItem($view, $entityName);
 
+        if (array_key_exists('export', $actions)) {
+            unset($actions['export']);
+        }
+
         if ($entityConfig['class'] === User::class) {
             if (!$this->authorizationChecker->isGranted('ROLE_SUPERADMIN')) {
                 $disabledActions = ['edit', 'delete'];
