diff --git a/src/AppBundle/Entity/GenusScientist.php b/src/AppBundle/Entity/GenusScientist.php
index 98b2ede..9a7002e 100644
--- a/src/AppBundle/Entity/GenusScientist.php
+++ b/src/AppBundle/Entity/GenusScientist.php
@@ -79,6 +79,10 @@ class GenusScientist
 
     public function __toString()
     {
-        return (string) $this->getUser();
+        return sprintf(
+            '%s studied genus for %d years',
+            (string)$this->getUser(),
+            $this->getYearsStudied()
+        );
     }
 }
