diff --git a/app/Resources/views/admin/fields/_id.html.twig b/app/Resources/views/admin/fields/_id.html.twig
index b204edd..38bd411 100644
--- a/app/Resources/views/admin/fields/_id.html.twig
+++ b/app/Resources/views/admin/fields/_id.html.twig
@@ -1 +1 @@
-<i class="fa fa-key"></i> {{ value }}
+{{ dump() }}<i class="fa fa-key"></i> {{ value }}
diff --git a/app/Resources/views/easy_admin/list.html.twig b/app/Resources/views/easy_admin/list.html.twig
index a9980c5..7e420b5 100644
--- a/app/Resources/views/easy_admin/list.html.twig
+++ b/app/Resources/views/easy_admin/list.html.twig
@@ -37,6 +37,7 @@
 {% block page_title %}{{ _content_title|striptags|raw }}{% endblock %}
 
 {% block content_header %}
+    {{ dump() }}
     <div class="row">
         <div class="col-sm-5">
             {% block content_title_wrapper %}
